### ISSUE TITLE
Principled stdlib support for child buffers and comint

### ISSUE THREAD

**Issue by matthewbauer** [shlevy/nix-buffer#13]
Currently the nix-buffer code in nixpkgs has very shaky support for detecting "child" buffers, and comint in particular is painful. We should have a better reusable version of this in some standard library.

Current description and title written by @shlevy

Original description and title from @matthewbauer :

# "compile" buffer doesn't inherit PATH or exec-path

This is a weird issue because I expected this to work.

Basically, running "compile" with something on nix-buffer's path will not be found for whatever reason. I think it is an issue with inheriting process-environment and exec-path.

**Comment by shlevy** [comment]
Yeah, I think we need a general solution for comint buffers... And really a better way to decide what is a "child" buffer than we currently have.
